src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -172,83 +159,139 @@ class ScatterPlot extends Visualization {
     /**
      * Adds panning and zooming functionality to the visualization.
      */
-    addPanAndZoom(box_width, box_height, svg, margin, scaleAndAxis, scatter, points) {
-        let zoomClass = 'zoom'
-        let minScale = 0.5
-        let maxScale = 20
+    addPanAndZoom(boxWidth, boxHeight, svg, margin, scaleAndAxis, scatter, points) {
+        const minScale = 0.5
+        const maxScale = 20
+        const rightButton = 2
+        const midButton = 1
+        const scrollWheel = 0
```

Comment:
Are these defined somewhere in D3 actually? If not maybe we should define them somewhere for use in all visualizations. Right now we just keep adding them in every file where we need them.

Reply:
No they're not, it would make sense to keep them in a single file, you're right.